src/Bicep.LangServer/Snippets/SnippetsProvider.cs

```diff
@@ -170,7 +170,7 @@ private IEnumerable<Snippet> GetRequiredPropertiesSnippetsForDisciminatedObjectT
 
     private string? GetSnippetText(TypeProperty typeProperty, int indentLevel, ref int index, string? discrimatedObjectKey = null)
     {
-        if (typeProperty.Flags.HasFlag(TypePropertyFlags.Required))
+        if (typeProperty.Flags.HasFlag(TypePropertyFlags.Required) && !TypeHelper.IsNullable(typeProperty.TypeReference.Type))
```

Comment:
Would it be better to have a TypeHelper.IsRequired() property that handles this logic?

Reply:
thx